src/foam/java/Constant.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.java',
+  name: 'Constant',
+
+  properties: [
+    'name',
+    'javaType',
```

Comment:
This is already java specific, so we just need 'type' with java being implied.
Also, we only need 'value', which is the 'javaValue'.